### PR TITLE
Refactor/sudoku page components

### PR DESCRIPTION
**Pull request by patrikglanemann**
I did some refactoring, some behind the scenes stuff so to say. The App still looks the same.

What I did:
- Create Components for Sudoku grid, individual rows and for submit(validate) button
- Create a custom hook to fetch data with method POST

For now I still store the initial Sudoku in local storage. It just didn't work to put it in a state. For some reason the state got updated, even thought I didn't do setState...

### CODE REVIEW

File: src/components/SudokuGrid/Row.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  mask,
+  onCellInRowClick,
+}) {
+  return (
+    <div className="Sudoku__row">
+      {currentSudokuNumbers[rowNumber].map((number, columnNumber) => {
+        let editable = true;
+        if (mask[rowNumber][columnNumber] !== 0) {
+          editable = false;
+        }
```
Comment (mpagels) [inline src/components/SudokuGrid/Row.js:16]: You could also do this
```suggestion
        const editable = mask[rowNumber][columnNumber] !== 0 ? false : true
```

And I would rename `editable` to `isEditable` <- the `is` indicates that the variable may hold a `boolean`

File: src/components/SudokuGrid/Row.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return (
+          <Cell
+            key={uuidv4()}
+            value={number}
+            id={[rowNumber, columnNumber]}
```
Comment (mpagels) [inline src/components/SudokuGrid/Row.js:21]: This is a weird looking `id`.

File: src/hooks/usePostFetch.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        ""
+      );
+
+    const encodeParams = (params) =>
+      Object.keys(params)
+        .map((key) => key + "=" + `%5B${encodeBoard(params[key])}%5D`)
+        .join("&");
+    const data = {
+      board: numbersArray,
+    };
```
Comment (mpagels) [inline src/hooks/usePostFetch.js:23]: This is super hard to read. First thought, that the `const data` was part of the `encodeParams` function. Would recommend putting an empty line in between.
```suggestion
    const encodeParams = (params) =>
      Object.keys(params)
        .map((key) => key + "=" + `%5B${encodeBoard(params[key])}%5D`)
        .join("&");
        
    const data = {
      board: numbersArray,
    };
```

**Review by mpagels — APPROVED** [review]
Very good refactoring, Patrik.
Found some small improvements though.

To your question. Is it not possible to create a `state` and set this state in the `useEffect`?

Something like this?

```js
export default function SudokuPage() {
  const [numbers, setNumbers] = useState([[], [], [], [], [], [], [], [], []]);
  const [isLoading, setIsLoading] = useState(false);
  const [sudokuStatus, setSudokuStatus] = useState("unsolved");
  const [sudoku, setSoduko] = useState([] // this is new

useEffect(() => {
    let url = "https://sugoku.herokuapp.com/board?difficulty=easy";
    setIsLoading(true);
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        setNumbers(data.board);
        setSoduko(data.board) // this is new
        localStorage.setItem("currentSudoku", JSON.stringify(data.board));
        setIsLoading(false);
      });
  }, []);

....// other code
}
```

File: src/components/SudokuGrid/Row.js (hunk 2 above)
Comment (patrikglanemann) [reply src/components/SudokuGrid/Row.js:21]: Hahah, yeah. Would it be bad practice to leave it like that, have an array as the id?

**Pull request merged**